miner/miner.go

```diff
@@ -0,0 +1,104 @@
+package miner
+
+import (
+	"errors"
+	"log"
+	"sync"
+
+	"go.sia.tech/core/chain"
+	"go.sia.tech/core/types"
+)
+
+type (
+	// A Miner mines blocks.
+	Miner interface {
+		MineBlock() types.Block
+		Address(addr types.Address) error
+	}
+
+	// A ChainManager manages blockchain state.
+	ChainManager interface {
+		AddTipBlock(b types.Block) error
+	}
+
+	// A Syncer can connect to other peers and synchronize the blockchain.
+	Syncer interface {
+		BroadcastBlock(block types.Block)
+	}
+)
+
+// MiningManager manages an internal block miner.
+type MiningManager struct {
```

Comment:
I'd rather keep this code in `main` like it was before. Eventually we'll want to design a fancier mining package (for pools), but I imagine that will look very different from a package designed for CPU mining.